src/biogeophys/CanopyFluxesMod.F90

```diff
@@ -865,13 +884,53 @@ subroutine CanopyFluxes(bounds,  num_exposedvegp, filter_exposedvegp,
       do f = 1, fn
          p = filterp(f)
          c = patch%column(p)
+         g = patch%gridcell(p)
+
+         select case (z0param_method)
+         case ('ZengWang2007')
+            lt = min(elai(p)+esai(p), tlsai_crit)
+            egvf =(1._r8 - alpha_aero * exp(-lt)) / (1._r8 - alpha_aero * exp(-tlsai_crit))
+            displa(p) = egvf * displa(p)
+            z0mv(p)   = exp(egvf * log(z0mv(p)) + (1._r8 - egvf) * log(z0mg(c)))
+
+         case ('Meier2022')
+            lt = max(0.00001_r8,elai(p)+esai(p))
+            displa(p) = htop(p) * (1._r8 - (1._r8 - exp(-(cd1_param * lt)**0.5_r8)) / (cd1_param*lt)**0.5_r8)
+
+            lt = min(lt,z0v_LAImax(patch%itype(p)))
+            delt = 2._r8
+            U_ustar_ini = (z0v_Cs(patch%itype(p)) + z0v_Cr(patch%itype(p)) * lt / 2._r8)**(-0.5_r8) &
+                      *z0v_c(patch%itype(p)) * lt / 4._r8
+            U_ustar = U_ustar_ini
+
+            do while (delt > 0.0001_r8)
+               U_ustar_prev = U_ustar
+               U_ustar = U_ustar_ini * exp(U_ustar_prev)
+               delt = abs(U_ustar - U_ustar_prev)
+            end do
+
+            U_ustar = 4._r8 * U_ustar / lt / z0v_c(patch%itype(p))
+
+            z0mv(p) = htop(p) * (1._r8 - displa(p) / htop(p)) * exp(-vkc * U_ustar + &
+                      log(z0v_cw(patch%itype(p))) - 1._r8 + z0v_cw(patch%itype(p))**(-1._r8))
+
+
+          case default
+            write(iulog,*) 'ERROR: unknown z0para_method: ', z0param_method
+            call endrun(msg = 'unknown z0param_method', additional_msg = errMsg(sourcefile, __LINE__))
+          end select
+
+          z0hv(p)   = z0mv(p)
+          z0qv(p)   = z0mv(p)
+
+          ! Update the forcing heights
+          ! TODO(KWO, 2022-03-15) Only for Meier2022 for now to maintain bfb with ZengWang2007
```

Comment:
Does this comment mean that we should remove the following if statement (here and elsewhere) going forward?

Reply:
Hmmm, good question!

Reply:
Who is KWO? That wouldn't be Ronny Meier... And it's not Keith Oleson either. We can't ask Ronny, but should look into this...

Reply:
@olyson do you know what's going on here with this comments and the other's like it?

Reply:
@swensosc do you know anything about this comment?

Reply:
That's me, W is my middle initial.  In the process of working on this PR I found that forcing heights weren't being updated properly, independent of the Meier changes.  But I wanted to maintain bfb for the original code (which uses ZengWang2007).  At some point (now?), yes, we will want to remove the if statement so that the forcing heights are updated for both Meier and ZengWang2007.

Reply:
Ahh, OK so Meier is doing the right thing, and we ought to change it so both are consistent. We probably should do that in a future update as an answer changing tag.

Reply:
We should make an issue to remind us to do that. @olyson could you start that one?

Reply:
To be clear, Meier is only doing the right thing because I added that code in.  I'll open an issue.

Reply:
I clarified with @ekluzek 
- We will merge with these lines as they are.
- @olyson will start an issue about it to update later.

Reply:
The issue is #2071 
So resolving.